src/many-ledger/src/storage.rs

```diff
@@ -1,16 +1,88 @@
 use crate::error;
+use many::message::ResponseMessage;
+use many::server::module;
 use many::server::module::abci_backend::AbciCommitInfo;
-use many::types::ledger::{Symbol, TokenAmount, Transaction, TransactionId, TransactionInfo};
-use many::types::{CborRange, SortOrder};
+use many::server::module::account;
+use many::server::module::account::features::multisig;
+use many::server::module::account::features::FeatureInfo;
+use many::types::ledger::{
+    AccountMultisigTransaction, Symbol, TokenAmount, Transaction, TransactionId, TransactionInfo,
+};
+use many::types::{CborRange, SortOrder, Timestamp};
 use many::{Identity, ManyError};
 use std::cmp::Ordering;
 use std::collections::{BTreeMap, BTreeSet, Bound};
 use std::ops::RangeBounds;
 use std::path::Path;
-use std::time::SystemTime;
+use std::time::{Duration, SystemTime};
 use tracing::info;
 
+const MULTISIG_ROLE_CAN_APPROVE: &str = "canMultisigApprove";
+const MULTISIG_ROLE_CAN_SUBMIT: &str = "canMultisigSubmit";
+
+fn _execute_multisig_tx(
+    ledger: &mut LedgerStorage,
+    _tx_id: &[u8],
+    storage: &TransactionStorage,
+) -> Result<Vec<u8>, ManyError> {
+    match &storage.info.transaction {
+        AccountMultisigTransaction::Send(module::ledger::SendArgs {
+            from: _from,
+            to,
+            symbol,
+            amount,
+        }) => ledger
+            .send(&storage.account, to, symbol, amount.clone())
+            .map(|_| vec![]),
+
+        AccountMultisigTransaction::AccountMultisigSubmit(arg) => ledger
+            .create_multisig_transaction(&storage.account, arg.clone())
+            .map(|_| vec![]),
+
+        AccountMultisigTransaction::AccountMultisigSetDefaults(arg) => ledger
+            .set_multisig_defaults(&storage.account, arg.clone())
+            .map(|_| vec![]),
+
+        _ => Err(multisig::errors::transaction_type_unsupported()),
+    }
+}
+
+#[derive(minicbor::Encode, minicbor::Decode)]
+#[cbor(map)]
+pub struct TransactionStorage {
```

Comment:
nit: I don't like the name; I feel like it's more of a `MultisigTransactionStorage`